synapse/storage/databases/main/push_rule.py

```diff
@@ -72,10 +72,10 @@ def _load_rules(
 
     ruleslist = [
         PushRule.from_db(
-            rule_id=rawrule["rule_id"],
-            priority_class=rawrule["priority_class"],
-            conditions=rawrule["conditions"],
-            actions=rawrule["actions"],
+            rule_id=rawrule[0],
+            priority_class=rawrule[1],
+            conditions=rawrule[2],
+            actions=rawrule[3],
```

Comment:
This is slightly less clear (and I should probably update the docstring), but not sure if matters much. Maybe we should return `PushRule` from the actual datastore methods?

Reply:
I think this is trickier than I initially thought... will give it another go though.

Reply:
If it doesn't work out, I think it's fine as-is.

Reply:
I'm going to add better docs to `_load_rules` to punt this, sorry.